testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -1184,6 +1193,22 @@ public void testRenameColumn()
         super.testRenameColumn();
     }
 
+    @Test
+    public void testInsertNegativeDate()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_INSERT));
```

Comment:
That means it's possible to have a connector that declares SUPPORTS_NEGATIVE_DATE, but it's not true, and there is no validation.

SUPPORTS_NEGATIVE_DATE should default to `SUPPORTS_INSERT || SUPPORTS_CREATE_TABLE_WITH_DATA` (without writes, negative dates cannot be supported)

then the test should verify the declaration is correct:

```java
if (!SUPPORTS_NEGATIVE_DATE) {
    if (!(SUPPORTS_INSERT || SUPPORTS_CREATE_TABLE_WITH_DATA)) {
      // nothing to test
     }
    checkState(SUPPORTS_INSERT); // TODO implement proper check for connectors with CTAS, but without INSERT
   do the negative test;
   return;
}

do the test
```